src/secure-proxy-handler.ts

```diff
@@ -232,14 +224,12 @@ export class SecureProxyHandler implements ProxyHandler<SecureProxyTarget> {
     setPrototypeOf(shadowTarget: SecureShadowTarget, prototype: SecureValue): boolean {
         this.initialize(shadowTarget);
         // this operation can only affect the env object graph
-        setPrototypeOf(shadowTarget, prototype);
-        return true;
+        return setPrototypeOf(shadowTarget, prototype);
     }
     preventExtensions(shadowTarget: SecureShadowTarget): boolean {
         // this operation can only affect the env object graph
         this.initialize(shadowTarget);
-        preventExtensions(shadowTarget);
-        return true;
+        return preventExtensions(shadowTarget);
```

Comment:
☝️ Here we return the actual value of `preventExtensions` and `setPrototypeOf` to avoid proxy trap invariant errors.

Reply:
is this the prevent extension from reflect?

Reply:
Yes.